### ISSUE TITLE
Generated clients are missing routing headers

### ISSUE THREAD

**Issue by busunkim96** [googleapis/gapic-generator-python#401]
https://aip.dev/client-libraries/4222

It looks like the generator is not adding variables declared in the `google.api.http` annotation to routing headers.

https://github.com/googleapis/googleapis/blob/52701da10fec2a5f9796e8d12518c0fe574488fe/google/cloud/servicedirectory/v1beta1/registration_service.proto#L60-L65

See https://github.com/GoogleCloudPlatform/python-docs-samples/pull/3456/files#r412305714 for the modifications that need to be made to currently use the library.

Possibly related to #380.

Metadata should be added [somewhere here](https://github.com/googleapis/python-service-directory/blob/25e0fcfb32d100be0cba3799d62543569dd2d2c6/google/cloud/servicedirectory_v1beta1/services/registration_service/client.py#L359-L378) (see [this method](https://github.com/googleapis/python-service-directory/blob/25e0fcfb32d100be0cba3799d62543569dd2d2c6/google/cloud/servicedirectory_v1beta1/services/registration_service/client.py#L432-L459) for comparison))

**Comment by software-dov** [comment]
Just to clarify, is this an issue being seen with microgenerated client libraries?

**Comment by busunkim96** [comment]
@software-dov Yep! [`google-cloud-service-directory`](https://github.com/googleapis/python-service-directory/blob/master/synth.py)

**Comment by rebeccawshaw** [comment]
Hi, is there any update on this bug? We need to get the python snippets in before our GA launch, as this is currently one of our blockers.

**Comment by busunkim96** [comment]
Bumping the priority on this as I'm seeing the same issue with KMS. 

Errors look like:

```
E           grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
E               status = StatusCode.NOT_FOUND
E               details = "The request concerns location 'us-east1' but was sent to location 'global'. Either Cloud KMS is not available in 'us-east1' or the request was misrouted. Make sure the 'x-goog-request-params' metadata is set correctly; see https://cloud.google.com/kms/docs/grpc for more information."
E               debug_error_string = "{"created":"@1591243631.459760738","description":"Error received from peer ipv4:74.125.142.95:443","file":"src/core/lib/surface/call.cc","file_line":1056,"grpc_message":"The request concerns location 'us-east1' but was sent to location 'global'. Either Cloud KMS is not available in 'us-east1' or the request was misrouted. Make sure the 'x-goog-request-params' metadata is set correctly; see https://cloud.google.com/kms/docs/grpc for more information.","grpc_status":5}"
E           >
```

**Comment by busunkim96** [comment]
The original issue with service_directory seems to be that the generator currently adds the metadata only for `http.get`. The methods missing metadata are of other types: `patch`, `delete`, `post`

```proto
  // Creates a namespace, and returns the new Namespace.
  rpc CreateNamespace(CreateNamespaceRequest) returns (Namespace) {
    option (google.api.http) = {
      post: "/v1beta1/{parent=projects/*/locations/*}/namespaces"
      body: "namespace"
    };
    option (google.api.method_signature) = "parent,namespace,namespace_id";
  }
```

https://github.com/googleapis/gapic-generator-python/blob/5f4c3009ec17399ab24a52a3b5bc9cbd0d9a2bf6/gapic/schema/wrappers.py#L624-L630

https://github.com/googleapis/gapic-generator-python/blob/41fa725a304792b3470d2be079e59f577387e5bb/gapic/templates/%25namespace/%25name_%25version/%25sub/services/%25service/client.py.j2#L331-L340

**Comment by software-dov** [comment]
Fixed by #443
